plugins/outputs/prometheus_remote_write/prometheus_remote_write.go

```diff
@@ -0,0 +1,136 @@
+package prometheus_remote_write
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/outputs/prometheus_client"
+)
+
+func init() {
+	outputs.Add("prometheus_remote_write", func() telegraf.Output {
+		return &PrometheusRemoteWrite{}
+	})
+}
+
+type PrometheusRemoteWrite struct {
+	URL           string `toml:"url"`
+	BasicUsername string `toml:"basic_username"`
+	BasicPassword string `toml:"basic_password"`
+}
+
+var sampleConfig = `
+  ## URL to send Prometheus remote write requests to.
+  url = "http://localhost/push"
```

Comment:
Add in the remaining options: https://github.com/influxdata/telegraf/wiki/SampleConfig